fdbserver/workloads/TenantManagementWorkload.actor.cpp

```diff
@@ -1719,6 +1723,28 @@ struct TenantManagementWorkload : TestWorkload {
 		}
 	}
 
+	ACTOR static Future<Void> readTenantKeys(TenantManagementWorkload* self) {
+		if (self->allTestTenants.size() == 0) {
+			return Void();
+		}
+		state Reference<Tenant> tenant = deterministicRandom()->randomChoice(self->allTestTenants);
+		state Reference<ReadYourWritesTransaction> tr = makeReference<ReadYourWritesTransaction>(self->dataDb, tenant);
+		loop {
+			try {
+				Optional<Value> val = wait(tr->get(self->keyName));
+				break;
+			} catch (Error& e) {
+				state Error err = e;
+				if (err.code() == error_code_tenant_not_found || err.code() == error_code_tenant_removed) {
```

Comment:
I'm not sure we will get `tenant_removed` here, but as for `tenant_not_found` and the values we read, it's important that we actually check that we get the expected result. That means we need to check specifically for these three things:

1. If the tenant exists and has a key, we read that key
2. If the tenant exists and doesn't have a key, we get an empty result
3. If the tenant doesn't exist, we get `tenant_not_found`

One way to do this is to create our `Tenant` objects with the name and ID, then read the tenant associated with the name stored in the `Tenant`. Then when we get here, we can read the associated object in `createdTenants`, and if it exists and matches the ID here we can use it.

Reply:
Are we actually getting `tenant_removed` here? If not, we should remove this.